RFCs/rfc-0007-zkPassport.md

```diff
@@ -0,0 +1,106 @@
+2. An o1js library that uses the Mina Attestation API Standard to create proofs of various components of one's zkPassport identity, for use in proof-of-unique-user and proof of nationality & age use cases.
+
+## Motivation and Rationale
+
+This would unlock many use cases for web3 and web2 applications connected to Mina.
+
+For applications on Mina, they can directly take advantage of this data through nullifiers for proof-of-uniqueness and other identity related proofs (see both scenarios below).
+
+For other chains, using Mina's zk state bridge, they could also leverage these proofs in their applications.
+
+Web2 applications may find use in this functionality as well. This seems particularly relevant as AI systems become increasingly powerful, and the risk this could pose to new and existing digital systems being overwhelmed by fake users.
+
+## Scenarios and Use Cases
+
+### Example Scenario 1: Proof-of-Uniqueness
+
+| Aspect           | Description |
+|------------------|-------------|
+| **Description**  | A zkApp wants a user to prove they are unique. |
+| **Requirements** | An implementation of zkPassport in the Attestation API Standard (see objectives), plus an on-chain nullifier that users can use with the zkApp. |
+| **Expected Outcome** | Via leveraging a simple library, users can make zkApps that are sybil-resistant. |
+| **Impact Analysis** | This would unlock many applications, such as one-person-one-vote, one-person-one account, Sybil-Resistant airdrops, etc. |
+
+### Example Scenario 2: Proof-of-nationality
+
+| Aspect           | Description |
+|------------------|-------------|
+| **Description**  | A zkApp wants a user to prove facts about what country they are / are not from. |
+| **Requirements** | An implementation of zkPassport in the Attestation API Standard (see objectives). |
+| **Expected Outcome** | Via leveraging a simple library, zkApps can require a user to prove facts about their nationality. |
+| **Impact Analysis** | This would be useful across web3, to make applications that are regulatory-compliant. |
+
+## Open Issues and Discussion Points
+
+A remaining question is exactly what new cryptography primitives would need to be implemented in o1js, if any new ones are needed. It looks like RSA, DSA, ECDSA, and Sha-xxx are used (see pages 15, 16, and 17 here[[4](https://www.icao.int/publications/Documents/9303_p12_cons_en.pdf)]).
+
+It seems countries keep databases of lost, stolen, and revoked travel documents[[5](https://www.icao.int/publications/Documents/9303_p2_cons_en.pdf)]. It doesn't seem like this is public however, so there may also have to be logic for invalidating one's old passport if one gets a new passport (due to the old one either being lost, stolen, revoked, or just expired).
```

Comment:
They have their own databases but it's not public. When passport is issued. The certification and data of it is valid on average 10years +3 months. So if they are stolen, lost etc. You can't use it to exit the country since they would know while you trying to leave. To my knowledge from NFC stand point there is no difference and I don't know any country with public DB on these documents.

Reply:
My understanding is that 10 years is the number due to expiration of NFC tags

Reply:
In Belgium citizens can log in to their government portal with eID card reader
https://www.youtube.com/watch?v=RBFe-FdPw34

Reply:
What you outlined here is eID not a passport. The 10 years is because passports are mostly valid for 10 years the problem is NFC is permanent the certificate is 10 years + few months so if it's stolen or something it will be always valid when you are checking it according to certificate.

Reply:
No, NFC expire in 10 years. I have seen this number on vendors pages, articles, etc

For example, 

> Finally, NFC tags could also be deleted if their data retention time expires. Data retention on NFC tags is intended to last for a set amount of time, typically 10 years. The tag might not be able to keep data properly if the data retention time has passed.

https://taptag.shop/blogs/how-tos/ways-an-nfc-tag-or-nfc-card-can-be-erased-magnets-heat-etc

Reply:
It does not matter if it is a drivers license, residence card, passport or eID NFC tags expire in typically 10 years

Reply:
https://taptag.shop/blogs/how-tos/what-is-the-difference-between-ntag213-ntag215-and-ntag216

> What is the difference between NTAG213, NTAG215, and NTAG216?
> 
> Number of scans: The NTAG213 can endure up to 50,000 scans, whereas the NTAG215 can withstand up to 200,000 scans. 500,000 scan operations are no problem for the NTAG216.
> 
> Operating frequency: The 13.56 MHz frequency is the same for all three tags.
> 
> Read/write distance: Depending on the NFC reader and the surrounding conditions, NTAG213, NTAG215 and NTAG216 all have a read/write distance of up to 4 cm.
> 
> Data retention: The 10-year data retention period applies to NTAG213, NTAG215, and NTAG216.
> 
> Data retention: The 10-year data retention period applies to NTAG213, NTAG215, and NTAG216.

Reply:
The NFC retention is totally depend on standard it utilize, there are NFC's which can hold data for 20-50 years etc. The topic we are talking here is validity of the document according to read we can get from NFC. The validity of the data is rely on certificate. Issue is when that document is invalid certificate or any data doesn't get updated because it's a passive system. You are just taking the conversation out of focus here in my opinion.

Reply:
Well, my passport has one of these tags, not the one that can hold data for 20-50 years etc

Reply:
I just think its quite extraordinary that we want to take in the account only the best case scenario and disregard all the rest, funny

Reply:
How long NFC store the data doesn't matter here. The important part is is it valid or enough to verify identity or not. I am saying it's not because certificate stored in NFC chip can be cloned or passport can be stolen and there will be no way you can check for that.

Reply:
Sure, but we can create an api service that helps in emergencies and that stores and is responsible for the data storage

Reply:
Here is NFC Tag Issuance Process from Near Field Communication (NFC) Technology with Jordan Mobile Payment (JoMoPay)
![image](https://github.com/MinaFoundation/Core-Grants/assets/66903336/53bd3955-8c08-4c50-984e-932a6e1f9393)

Reply:
Ideally, any architecture makes it possible to override a previously registered passport with a newer one. That seems to be an important fallback flow in case your identity is stolen. If that is possible, it doesn't matter how long the original passport or NFC tag is valid

Reply:
I don't think that covers cloning or stolen at all. As random people from earth's population. I will never learn my identity utilized on mina blockchain in the first place.

Reply:
I agree, it doesn't cover that (a liveness  test is also needed), but in an ideal system you can still override an old passport + liveness test with a newer one, and so it doesn't matter how long the old one is valid